components/ValueTypeDecoder/ValueTypeDecoder.tsx

```diff
@@ -1,77 +1,130 @@
 /**
  * @author Hugo Masclet <git@hugom.xyz>
  */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { ERC725, ERC725JSONSchema } from '@erc725/erc725.js';
-import { Erc725JsonSchemaAll } from '../../interfaces/erc725';
 import AddressButtons from '../AddressButtons';
 import { LUKSO_IPFS_BASE_URL } from '../../globals';
+import Web3 from 'web3';
+
+import { DecodeDataOutput } from '@erc725/erc725.js/build/main/src/types/decodeData';
 
 interface Props {
-  erc725JSONSchema: ERC725JSONSchema | Erc725JsonSchemaAll;
-  value: string;
+  provider: Web3;
+  address: string;
+  erc725JSONSchema: ERC725JSONSchema;
+  value: string | string[];
 }
 
-const ValueTypeDecoder: React.FC<Props> = ({ erc725JSONSchema, value }) => {
-  if (erc725JSONSchema.valueContent === 'Address') {
-    return (
-      <>
-        <code>{value}</code>
-        <AddressButtons address={value} />
-      </>
-    );
-  }
+const ValueTypeDecoder: React.FC<Props> = ({
+  provider,
+  address,
+  erc725JSONSchema,
+  value,
+}) => {
+  const [decodedDataOneKey, setDecodedDataOneKey] = useState<{
+    [key: string]: any;
+  }>([]);
+  const [fetchedData, setFetchedData] = useState<DecodeDataOutput>({
+    key: '',
+    name: '',
+    value: [],
+  });
 
-  // The schema may be wrong, this error will be catched bellow
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
-  const schema: ERC725JSONSchema[] = [erc725JSONSchema];
+  useEffect(() => {
+    const startDecoding = async () => {
+      if (address) {
+        const schema: ERC725JSONSchema[] = [erc725JSONSchema];
+        const erc725 = new ERC725(schema, address, provider.currentProvider);
 
-  const erc725 = new ERC725(schema);
+        const decodedData = erc725.decodeData([
+          {
+            keyName: erc725JSONSchema.name,
+            value: value as string,
+          },
+        ]);
 
-  let decodedDataOneKey;
-  try {
-    decodedDataOneKey = erc725.decodeData({
-      [erc725JSONSchema.name]: value,
-    });
-  } catch (err) {
-    // Goes here if key is not in erc725.js yet or if key is undefined
-    return <span>Can&apos;t decode this key</span>;
-  }
+        setDecodedDataOneKey(decodedData);
+
+        const result = await erc725.getData(erc725JSONSchema.name);
+        console.log(result);
+        setFetchedData(result);
+      }
+    };
+    startDecoding();
+  }, [address]);
 
   try {
-    return (
-      <div>
-        <pre>
-          {JSON.stringify(decodedDataOneKey[erc725JSONSchema.name], null, 4)}
-        </pre>
-        {decodedDataOneKey[erc725JSONSchema.name].url && (
+    if (typeof decodedDataOneKey[0].value === 'string') {
+      if (erc725JSONSchema.valueContent === 'Address') {
+        return <AddressView value={decodedDataOneKey[0].value} />;
```

Comment:
not sure if it makese sense to create an extra component bellow, as this is only used here? Why not directly put the html here? (only 2 lines....)